### PR TITLE
chore: don't use XL runners

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4785" title="WPB-4785" target="_blank">TaskWPB-4785</a>  Use XL machines for running tests
  </td></table>
  <br />
 

This reverts commit 162f5641c13732c05a77045a919c096765a5cb9e.

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Reverting because XL runners are not included in our monthly minutes but are billed separately.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/swiftlint.yml [hunk 1]
```diff
@@ -19,5 +19,4 @@ jobs:
         - uses: actions/checkout@v4
         - name: GitHub Action for SwiftLint
           uses: norio-nomura/action-swiftlint@3.2.1
-          with:
-            args: --strict
```
Comment (agisilaos) [inline .github/workflows/swiftlint.yml]: Can we keep this?

**Review by agisilaos — APPROVED** [review]

File: .github/workflows/swiftlint.yml (hunk 1 above)
Comment (johnxnguyen) [reply .github/workflows/swiftlint.yml]: let's add it in a separate PR

File: .github/workflows/_reusable_run_tests.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ on:
 
 jobs:
   run-tests:
-    runs-on: macos-13-xl
+    runs-on: macos-13
```
Comment (netbe) [inline .github/workflows/_reusable_run_tests.yml:76]: just revert this!

File: .github/workflows/_reusable_run_tests.yml [hunk 3]
```diff
@@ -308,12 +308,26 @@ jobs:
...
+        with:
+          path: fastlane/.post_build
+
       - name: Notify on Wire if failed
         if: ${{ failure() && github.ref_name == 'develop' }}
         uses: 8398a7/action-slack@v3
         with:
           status: ${{ job.status }}
-          text: "🆘 Tests for 'develop' failed 🆘\ncommit: https://github.com/wireapp/wire-ios-mono/commit/${{ github.sha }}\n**Triggered by:** ${{ github.triggering_actor }}\n**Build log:** ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}\n"
+          text: "🆘 Tests for 'develop' failed for commit: *${{ github.sha }}* 🆘\n**Triggered by:** ${{ github.triggering_actor }}\n**Build log:** ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}\n"
```
Comment (netbe) [inline .github/workflows/_reusable_run_tests.yml]: this needs to be kept

File: .github/workflows/_reusable_run_tests.yml (hunk 2 above)
Comment (johnxnguyen) [reply .github/workflows/_reusable_run_tests.yml:76]: yep good point!

File: .github/workflows/_reusable_run_tests.yml (hunk 3 above)
Comment (johnxnguyen) [reply .github/workflows/_reusable_run_tests.yml]: fixed

**Review by netbe — APPROVED** [review]

**Pull request merged**